jumpscale/packages/vdc/services/check_threebot.py

```diff
@@ -0,0 +1,85 @@
+from jumpscale.loader import j
+
+from jumpscale.sals.zos import get as get_zos
+from jumpscale.tools.servicemanager.servicemanager import BackgroundService
+from jumpscale.clients.explorer.models import WorkloadType
+
+
+class CheckThreebot(BackgroundService):
+    def __init__(self, interval=60 * 60, *args, **kwargs):
+        """Check that Threebot Container is UP and Redeploy if DOWN
+        """
+        super().__init__(interval, *args, **kwargs)
+
+    def job(self):
+        zos = get_zos()
+        threebot_workload_types = [WorkloadType.Container, WorkloadType.Subdomain]
+        j.logger.info("Check VDC threebot service: Check if Threebot containers in all VDCs is UP")
+        for vdc_name in j.sals.vdc.list_all():
```

Comment:
you need to yield control between each vdc using gevent.sleep

Reply:
Updated